ingestion/transformers/assign_cadet_domains.py

```diff
@@ -0,0 +1,121 @@
+    manifest_s3_uri: str
+
+
+class AssignDerivedTableDomains(AddDatasetDomain):
+    """Transformer that adds a specified domains to each dataset."""
+
+    def __init__(
+        self, config: CadetDatasetDomainSemanticsConfig, ctx: PipelineContext
+    ):
+        AddDatasetDomain.raise_ctx_configuration_error(ctx)
+        manifest = self._get_manifest(config)
+        domain_mappings = self._get_domain_mapping(manifest)
+        domain_pattern = domain_mappings.domain_pattern
+
+        def resolve_domain(domain_urn: str) -> DomainsClass:
+            domains = domain_pattern.value(domain_urn)
+            return self.get_domain_class(ctx.graph, domains)
+
+        generic_config = AddDatasetDomainSemanticsConfig(
+            get_domains_to_add=resolve_domain,
+            semantics=config.semantics,
+            replace_existing=config.replace_existing,
+        )
+        super().__init__(generic_config, ctx)
+
+    @classmethod
+    def create(
+        cls, config_dict, ctx: PipelineContext
+    ) -> "AssignDerivedTableDomains":
+        try:
+            manifest_s3_uri = config_dict.get("manifest_s3_uri")
+            replace_existing = config_dict.get("replace_existing", False)
+        except Exception as e:
+            print(e)
+            raise
+
+        config_dict = CadetDatasetDomainSemanticsConfig(
+            manifest_s3_uri=manifest_s3_uri,
+            replace_existing=replace_existing,
+        )
+        return cls(config_dict, ctx)
+
+    def _get_manifest(self, config_dict: CadetDatasetDomainSemanticsConfig) -> Dict:
+        try:
+            s3 = boto3.client("s3")
+            s3_parts = config_dict.manifest_s3_uri.split("/")
+            bucket_name = s3_parts[2]
+            file_key = "/".join(s3_parts[3:])
+            response = s3.get_object(Bucket=bucket_name, Key=file_key)
+            content = response['Body'].read().decode('utf-8')
+            manifest = json.loads(content, strict=False)
+        except NoCredentialsError:
+            print("Credentials not available.")
+            raise
+        except ClientError as e:
+            # If a client error is thrown, it will have a response attribute containing the error details
+            error_code = e.response['Error']['Code']
+            print(f"Client error occurred: {error_code}")
+            raise
+        except json.JSONDecodeError:
+            print("Error decoding manifest JSON.")
+            raise
+        except Exception as e:
+            # Catch any other exceptions
+            print(f"An error occurred: {str(e)}")
+            raise
+        return manifest
+
+    def _get_domain_mapping(self, manifest) -> PatternDatasetDomainSemanticsConfig:
+        """Map regex patterns for tables to domains"""
+        nodes = manifest.get("nodes")
+        domain_mappings = {}
+
+        for node in nodes:
+            if manifest["nodes"][node]["resource_type"] == "seed":
```

Comment:
If sources don't need domains we can add "source" here as well. This will reduce the size of the mapping dict so speed up ingestion